API/users/models.py

```diff
@@ -1,2 +1,45 @@
+from django.db import models
+from django.contrib.auth.models import User
 
 # Create your models here.
+
+class Account(models.Model):
+    name = models.CharField(max_length=64)
+    user = models.ForeignKey(User, on_delete=models.CASCADE)
+    def __str__(self):
+        return self.name
+
+class Transaction(models.Model):
+    account = models.ForeignKey(Account, on_delete=models.CASCADE)
+    timestamp = models.DateTimeField()
+    def __str__(self):
+        return self.name
+
+class Category(models.Model):
+    name = models.CharField(max_length=64)
+    parent = models.ForeignKey('self', null=True, on_delete=models.SET_NULL)
+    def __str__(self):
+        return self.name
+
+class ProductGroup(models.Model):
+    owner = models.ForeignKey(User, on_delete=models.CASCADE)
+    account = models.ManyToManyField(Account)
+    def __str__(self):
+        return self.name
+
+class Product(models.Model):
+    name = models.CharField(max_length=64)
+    category = models.ForeignKey(Category, null=True, on_delete=models.SET_NULL)
+    group = models.ForeignKey(ProductGroup, on_delete = models.CASCADE)
+    def __str__(self):
+        return self.name
+
+class TransactionItem(models.Model):
+    quantity = models.FloatField()
+    price = models.FloatField()
+    product = models.ForeignKey(Product, on_delete=models.CASCADE)
```

Comment:
I am not sure about `CASCADE` here. I can imagine a scenario where `Product` record gets deleted by the user. In this case we would probably like to keep `TransactionItem` in database. Maybe change it to `SET_NULL`?